trismart_smartboard_integration/models/smartboard_connector.py

```diff
@@ -0,0 +1,42 @@
+from odoo import api, fields, models, tools, SUPERUSER_ID
+
+class SmartBoard(models.Model):
+    _name = "smartboard.connector"
+    _description = "Create SB Project"
+
+    @api.model
+    def create_project(self, sb_lead_id):
+        try:
+            if len(sb_lead_id):
+                partner_exist = self.env['res.partner'].search([('sb_lead_id', '=', sb_lead_id[0])], limit=1)
+                lead_exist = self.env['crm.lead'].search([('sb_lead_id', '=', sb_lead_id[0])], limit=1)
+                project_exist = self.env['project.project'].search([('sb_lead_id', '=', sb_lead_id[0])], limit=1)
+
+                if not partner_exist:
+                    # Create new partner
+                    partner_id = self.env['res.partner'].create({'name': 'SBLead-' + sb_lead_id[0], 'sb_lead_id': sb_lead_id[0], "sync_status" : '1'})
+                if not lead_exist:
+                    # Create new lead
+                    lead_id = self.env['crm.lead'].create({'name': 'SBLead-' + sb_lead_id[0], 'sb_lead_id': sb_lead_id[0]})
+
+                    # Convert lead to opportunity
+                    customer = self.env['res.partner'].browse(partner_id) if partner_id else partner_exist
+                    for lead in lead_id:
+                        vals = lead._convert_opportunity_data(customer, team_id=False)
+                        lead.write(vals)
+
+                if not project_exist:
+                    # Create new project
+                    project_id = self.env['project.project'].create({'name': 'SBLead-' + sb_lead_id[0], 'sb_lead_id': sb_lead_id[0]})
+                    project_id.partner_id = partner_id
+
+                if partner_exist:
+                    return {"status": 200,  "message": "partner exists", "odoo_customer_id": []}
+                else:
+                    return {"status": 200, "message": "partner created", "odoo_customer_id": partner_id.id}
```

Comment:
The revised code will only return when partner is created

Reply:
If the partner existed, shouldn't they have the customer_id too?

Reply:
I revised the code again, will return when partner exists or partner created. I will leave the error blank as ticket description. If partner exists then the odoo_customer_id will be false to prevent confusion for user. The odoo_customer_id indicates the successfully creation of new partner.